index.js

```diff
@@ -39,20 +39,78 @@ const SUPPORTS_MULTIPLE_PLACEMENTS = false;
 
 /* Placement object to query decision API and return an Element node
  *
- * @param publisher
- * @param ad_type
+ * @param {string} publisher - Publisher ID
+ * @param {string} ad_type - Placement ad type id
+ * @param {Element} target - Target element
  */
 export class Placement {
-  constructor(publisher, ad_type = "image") {
+  constructor(publisher, ad_type = "image", target) {
     this.publisher = publisher;
     this.ad_type = ad_type;
+    this.target = target;
   }
 
-  /* Load placement from decision API and create DOM element
+  /* Create a placement from an element
    *
-   * @returns Promise
+   * @static
+   * @param {Element} element - Load placement and append to this Element
+   * @returns {Placement}
+   */
+  static from_element(element) {
+    // Get attributes from DOM node
+    const publisher = element.getAttribute(ATTR_PREFIX + "publisher");
+    let ad_type = element.getAttribute(ATTR_PREFIX + "type");
+    if (!ad_type) {
+      ad_type = "image";
+      element.setAttribute(ATTR_PREFIX + "type", "image");
+    }
+
+    // Add version to ad type to verison the HTML return
+    ad_type += "-v" + AD_CLIENT_VERSION;
+
+    return new Placement(publisher, ad_type, element);
+  }
+
+  /* Transforms target element into a placement
+   *
+   * This method organizes all of the operations to transform the placement
+   * configuration wrapper `div` into an ad placement -- including starting the
+   * API transaction, displaying the ad element, and eventually handling the
+   * viewport detection.
+   *
+   * @returns {Promise}
    */
   load() {
+    return this.fetch().then((element) => {
+      // If element is not defined, it's likely the underlying API call was
+      // blocked by an ad blocker. Don't show an error in this case, but tuck
+      // away an error in the DOM for debug purposes.
+      if (element === undefined) {
+        this.target.setAttribute(
+          ATTR_PREFIX + "error",
+          "Ad decision request blocked"
+        );
+        return;
+      }
+
+      // Add `loaded` class, signifying that the CSS styles should finally be
+      // applied to the target element.
+      let classes = this.target.className || "";
+      classes += " loaded";
+      this.target.className += classes.trim();
+      this.target.appendChild(element);
```

Comment:
I think we should consider replacing what's inside the element. Some people might use the space for a fallback placement or to display a placeholder.